tests/e2e/test_load.py

```diff
@@ -0,0 +1,60 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import torch
+import ssl
+import functools
+import certifi
+
+from azstoragetorch.io import BlobIO
+
+
+@pytest.fixture(scope="module")
+def ssl_inject_certifi(monkeypatch):
+    # Sets the global variable for the SSL certificates file path since torch.hub doesn't use the certifi package
+    monkeypatch.setattr(
+        ssl,
+        "_create_default_https_context",
+        functools.partial(ssl._create_default_https_context, cafile=certifi.where()),
+    )
+
+
+@pytest.fixture(scope="module")
+def tmp_path(tmp_path_factory):
+    return tmp_path_factory.mktemp("torch_hub")
+
+
+@pytest.fixture(scope="module")
+def model(tmp_path):
+    torch.hub.set_dir(tmp_path)
+    model = torch.hub.load("pytorch/vision:v0.10.0", "resnet101", pretrained=False)
+    return model
+
+
+@pytest.fixture(scope="module")
+def upload_model(model, container_client):
+    blob_name = "model.pth"
+    torch.save(model.state_dict(), blob_name)
```

Comment:
We should probably use the [`tmp_path`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.tmpdir.tmp_path) fixture for the name of the file we save the model to. Right now it looks like the `model.pth` is being saved to the current working directory of our tests, which means it won't be deleted after cleanup.

Another option is we can just provide a [`BytesIO`](https://docs.python.org/3/library/io.html#io.BytesIO) object to `torch.save()` and use that directly when uploading. The nice part about this is that we would not be writing to disk and possibly leak state between test runs. I don't have a strong preference though on which option we go with.